coffee-model/coffee-model-base/src/main/java/hu/icellmobilsoft/coffee/model/base/ClassFieldsAndMethodsCache.java

```diff
@@ -0,0 +1,115 @@
+/*-
+ * #%L
+ * Coffee
+ * %%
+ * Copyright (C) 2020 i-Cell Mobilsoft Zrt.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.model.base;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+import jakarta.enterprise.inject.Vetoed;
+
+/**
+ * Cache for fields and methods of entity classes used in {@link AbstractProvider}
+ *
+ * @author zsolt.vasi
+ * @since 2.0.0
+ */
+@Vetoed
+public class ClassFieldsAndMethodsCache {
+
+    private static final int DEFAULT_CACHE_SIZE = 10_000;
+
+    private final LoadingCache<Class<?>, Pair<List<Field>, List<Method>>> cache;
```

Comment:
elgondolkodtam hogy csak ez miatt kell-e ide guava. Jelen pillanatban eleg ide egy egyszeru cache megoldas is... Tobbiek mit gondoltok?

Reply:
en kihagynam a guavat, fuggoseg nelkul egyszeruen megoldhatjuk

Reply:
Lehet, hogy túlzás csak a cache miatt bekötni a guava-t. Ha csak cache kell van arra külön lib https://github.com/ben-manes/caffeine.

Ha pedig csak egy méret korlátos LRU cache kell, azt egyszerűen egy LinkedHashMap-pel is meg lehet oldani

Reply:
Ok, keresek egy egyszerűbb megoldást. Esetleg azt majd át lehetne gondolni, hogy máshol ahol guava van (pl coffee-tool RegexPatternCache esetén) szükséges-e, vagy ott is elengedhető a függőség

Reply:
Lecseréltem LinkedHashMap-re.